### DISCUSSION TITLE
New Image Component

### DISCUSSION THREAD

**Discussion by liornz** [vercel/next.js#44409] [Help]
The old Image Component had a prop layout="responsive".
How do I achieve this with the new version 13 Image Component?
These are images that I cannot import statically,  because it's a Card component that gets the path as props.
I am using the fill prop but since the cards are in a grid the size of the card changes depending on on how many cards fit in the viewport.

**Comment by ciruz** [comment]
Does this answer your question?

With next.js ``13.x`` you can now simply use the ``fill`` attribute, like so:

```javascript
<Image src={image} alt="" fill />
```

result:
![image](https://user-images.githubusercontent.com/987947/209806162-69db51a9-6229-44fb-8631-4301007fcfb0.png)


**If you want to style it, the ``parent`` element must have ``position: relative`` and ``display: block``.**

If you want to control the size of the image, you use the parent element with fill, like so:
```javascript
<div
  style={{
    position: "relative",
    width: "300px",
    height: "100px",
    border: "solid 3px blue",
  }}
>
    <Image src={image} alt="" fill />
</div>
```

result:
![image](https://user-images.githubusercontent.com/987947/209806616-9df49be7-87f0-4de6-b553-81e62ab6cb4d.png)


Or if this is not the desired look, use ``object-fit`` now via ``css/style`` instead of an attribute, like it was the case with next 12.

```javascript
<div
  style={{
    position: "relative",
    width: "300px",
    height: "100px",
    border: "solid 3px blue",
  }}
>
    <Image src={image} alt="" fill style={{ objectFit: "cover" }} />
</div>
```

result:
![image](https://user-images.githubusercontent.com/987947/209806834-aea14a83-07ae-4833-bda6-e6e022fff0db.png)

**Reply by kendjr** [reply]
@ciruz is it possible for you to look at [this post](https://github.com/vercel/next.js/discussions/44486)? The height on my images is always zero, unless I put another element inside of the container with the image element, and then it just takes the height of that element.
